### PR TITLE
Feature/int 734 add create pr UI to the webview

### PR DESCRIPTION
**Pull request by DmytroHryshyn**
Added  UI for creating Pull Request from extension
image

### CODE REVIEW

File: intuita-webview/src/App.tsx [hunk 1]
```diff
@@ -74,36 +78,37 @@ function App() {
...
+			message="In order to create issues, link your Intuita account"
+			actionButtons={[
+				<VSCodeButton onClick={handleLinkAccount}>
+					Link account{' '}
+				</VSCodeButton>,
+			]}
+		/>;
+	}
+
+	if (!view) return null;
```
Comment (grzpab) [inline intuita-webview/src/App.tsx:105]: `{ return null; }`

File: src/extension.ts [hunk 2]
```diff
@@ -215,6 +249,19 @@ export async function activate(context: vscode.ExtensionContext) {
 		}),
 	);
 
+	context.subscriptions.push(
+		vscode.commands.registerCommand(
+			'intuita.sourceControl.createPR',
+			async () => {
+				try {
+					sourceControl.createPR();
```
Comment (grzpab) [inline src/extension.ts:257]: is it awaitable?

File: src/components/webview/sourceControl.ts [hunk 3]
```diff
@@ -20,6 +20,15 @@ export class SourceControlService {
 		private readonly __messageBus: MessageBus,
 	) {}
 
+	// @TODO
```
Comment (grzpab) [inline src/components/webview/sourceControl.ts:23]: do you know how we could get that?

File: src/components/webview/sourceControl.ts [hunk 4]
```diff
@@ -20,6 +20,15 @@ export class SourceControlService {
...
+	getBaseBranchName() {
+		return 'MainBranchName';
+	}
+
+	async createPR() {
```
Comment (grzpab) [inline src/components/webview/sourceControl.ts:28]: that's just a PR for the UI right?

File: src/components/webview/IntuitaPanel.ts [hunk 5]
```diff
@@ -54,6 +58,43 @@ export type WebviewResponse =
...
+				baseBranch: string;
+				targetBranch: string;
+			};
+	  }>;
+
+export type ViewId = 'createIssue' | 'createPR';
+
+export type View =
+	| Readonly<{
+			viewId: ViewId & 'createIssue';
```
Comment (grzpab) [inline src/components/webview/IntuitaPanel.ts:76]: why this trick?

File: package.json [hunk 6]
```diff
@@ -53,6 +53,10 @@
 			}
 		],
 		"commands": [
+			{
+				"command": "intuita.createPR",
+				"title": "Intuita:create pr"
```
Comment (grzpab) [inline package.json:58]: Create a Pull Request

File: package.json [hunk 7]
```diff
@@ -156,6 +160,11 @@
 				"command": "intuita.createIssue",
 				"title": "Create Issue",
 				"icon": "$(github)"
+			},
+			{
+				"command": "intuita.createPR",
+				"title": "Create PR",
```
Comment (grzpab) [inline package.json:166]: Create a Pull Request

File: src/components/webview/sourceControl.ts (hunk 4 above)
Comment (DmytroHryshyn) [reply src/components/webview/sourceControl.ts:28]: Yes, this PR is just UI, so methods in services are empty

File: src/components/webview/sourceControl.ts (hunk 3 above)
Comment (DmytroHryshyn) [reply src/components/webview/sourceControl.ts:23]: This should work https://stackoverflow.com/questions/28666357/how-to-get-default-git-branch
I will implement it in next PR

File: src/components/webview/IntuitaPanel.ts (hunk 5 above)
Comment (DmytroHryshyn) [reply src/components/webview/IntuitaPanel.ts]: fixed

File: intuita-webview/src/App.tsx [hunk 8]
```diff
@@ -74,36 +77,39 @@ function App() {
...
+			/>
+		);
+	}
+
+	if (!linkedAccount) {
+		<WarningMessage
+			message="In order to create issues, link your Intuita account"
+			actionButtons={[
+				<VSCodeButton onClick={handleLinkAccount}>
+					Link account{' '}
```
Comment (grzpab) [inline intuita-webview/src/App.tsx:98]: can you remove `{' '}`?

**Review by grzpab — APPROVED** [review]

**Pull request merged**